src/Compilers/Core/Portable/Emit/SemanticEdit.cs

```diff
@@ -154,11 +178,15 @@ public SemanticEdit(SemanticEditKind kind, ISymbol? oldSymbol, ISymbol? newSymbo
             Kind = kind;
             OldSymbol = oldSymbol;
             NewSymbol = newSymbol;
-            PreserveLocalVariables = preserveLocalVariables;
             SyntaxMap = syntaxMap;
             Instrumentation = instrumentation;
         }
 
+        /// <summary>
+        /// True if <see cref="SyntaxMap"/> is not null.
+        /// </summary>
+        public bool PreserveLocalVariables => SyntaxMap != null;
```

Comment:
nit: this could be annotated with `[MemberNotNullWhen(returnValue: true, nameof(SyntaxMap))]`